Subscription/Subscription.sol

```diff
@@ -122,6 +122,14 @@ contract Subscription {
         view
         returns (bytes32)
     {
+        // if there are requirements from the deployer, let's make sure
```

Comment:
Added these require statements to `getSubscriptionHash` because this placements prevents any subscription from ever being created that not doe snot meet the requirements of the contract. Also, I was able to remove these same checks elsewhere in the contract because `getSubscriptionHash` was also being called in these places and now performs the check.